### PR TITLE
use handle_wait in pick behavior

### LINKED ISSUE #387: [ctitical] implement cob_default_robot_behaviour using handles
especially for torso movements and pick movements, we should check every command using the handles and abort if a previous movement cannot be completed as expected! 

```
handle = sss.move()
handle.wait()
if handle.get_error_code() != 0:
   ABORT
```
- define a handle for each `sss.move` command
- wait for movements to be finished before executing new movements
- do NOT use `spleep()` to synchronize movements (which can cause a collission)
- only continue if error_code is zero (==0)

This is currently missing for pick! ...disabling this button for now in the robot dashboard!

### PR DESCRIPTION
**Pull request by fmessmer**
fixes https://github.com/ipa320/cob_robots/issues/387

needs to be tested on robot.

### CODE REVIEW

**Comment by fmessmer** [comment]
Using `handle.wait()` allows for **safer** execution, i.e. each motion needs to be successful before next motion start...abort with failure otherwise. 
However, the behavior now (might) look like stop-and-go.
For the case where **fluent** motion is required, I kept the `sleep()`-synchronization in comments. Simply bring it back in and comment out the "wait-if-return" blocks...

**Comment by NadiaHG** [comment]
Already tested on the robot?

**Comment by fmessmer** [comment]
No...as robot currently is mostly "armless"... :wink:

**Comment by fmessmer** [comment]
@ipa-nhg please combine this PR with #591 in order to fix #387

**Comment by fmessmer** [comment]
includes #591

**Review by floweisshardt — APPROVED** [review]

File: cob_default_robot_behavior/scripts/trigger_srvs_cob4-7.py [hunk 1]
```diff
@@ -67,13 +67,132 @@
...
+    handle_arm_left.wait()
+    handle_arm_right.wait()
+    if not(handle_arm_left.get_error_code() == 0 and handle_arm_right.get_error_code() == 0):
+        return TriggerResponse(False, "Could not move arms.")
+
+    rospy.loginfo("------------------torso movement")
+    handle_torso = sss.move("torso","front_down_full",False)
+    handle_base = sss.move_base_rel("base",[0,0,1.57],False)
+    handle_torso.wait()
+    handle_base.wait()
```
Comment (floweisshardt) [inline cob_default_robot_behavior/scripts/trigger_srvs_cob4-7.py]: error while testing on robot (`move_base_rel` does not support `.wait`)

```
[INFO] [WallTime: 1485193296.888138] Set <<mimic>> to <<['happy', 0, 1]>>
[INFO] [WallTime: 1485193296.989465] Move <<arm_left>> to <<side>>
[INFO] [WallTime: 1485193297.277909] Move <<arm_right>> to <<side>>
[INFO] [WallTime: 1485193297.604288] Wait for <<arm_left>> reaching <<side>>...
[INFO] [WallTime: 1485193297.604746] ...<<arm_left>> reached <<side>>
[INFO] [WallTime: 1485193297.605281] Wait for <<arm_right>> reaching <<side>>...
[INFO] [WallTime: 1485193297.705841] ...<<arm_right>> reached <<side>>
[INFO] [WallTime: 1485193297.706162] ------------------torso movement
[INFO] [WallTime: 1485193297.708325] Move <<torso>> to <<front_down_full>>
[INFO] [WallTime: 1485193297.962450] Move base relatively by <<[0, 0, 1.57]>>
[INFO] [WallTime: 1485193297.964314] Wait for <<torso>> reaching <<front_down_full>>...
[INFO] [WallTime: 1485193298.065064] ...<<torso>> reached <<front_down_full>>
[INFO] [WallTime: 1485193298.065414] Wait for <<base>> reaching <<[0, 0, 1.57]>>...
[ERROR] [WallTime: 1485193298.066156] Error processing request: 'NoneType' object has no attribute 'wait_for_result'
['Traceback (most recent call last):\n', '  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 623, in _handle_request\n    response = convert_return_to_response(self.handler(request), self.response_class)\n', '  File "/u/fmw/git/care-o-bot/src/cob_robots/cob_default_robot_behavior/scripts/trigger_srvs_cob4-7.py", line 109, in pick_cb\n    handle_base.wait()\n', '  File "/u/robot/git/care-o-bot/install/lib/python2.7/dist-packages/simple_script_server/simple_script_server.py", line 1220, in wait\n    self.wait_for_finished(duration,True)\n', '  File "/u/robot/git/care-o-bot/install/lib/python2.7/dist-packages/simple_script_server/simple_script_server.py", line 1251, in wait_for_finished\n    self.client.wait_for_result()\n', "AttributeError: 'NoneType' object has no attribute 'wait_for_result'\n"]
```

**Review by floweisshardt — CHANGES_REQUESTED** [review]

**Comment by floweisshardt** [comment]
sorry for creating a merge conflict again by merging #591

**Comment by fmessmer** [comment]
I rebased...but don't know how to solve `move_base_rel`....
Even more, I think #591 has some issues, too - see https://github.com/ipa320/cob_robots/pull/591/files#r97415187

**Comment by fmessmer** [comment]
How is this?
Could you test it again?
All behaviors are harmonized now...

**Comment by fmessmer** [comment]
finally remove pick as it was never used again
-> #636

**Pull request closed without merging**